firstvoices/backend/tasks/import_job_tasks.py

```diff
@@ -215,6 +218,72 @@ def import_audio_resource(import_job, audio_data, dictionary_entry_data, dry_run
     return audio_import_result, dictionary_entry_data
 
 
+def import_img_resource(import_job, img_data, dictionary_entry_data, dry_run):
+    """
+    Imports image files and appends IDs of the imported files as related_images in dictionary_entry_data.
+    Returns updated dictionary_entry_data and result from image import.
+    """
+
+    img_import_result = ImageResource(
+        site=import_job.site,
+        run_as_user=import_job.run_as_user,
+        import_job=import_job.id,
+    ).import_data(dataset=img_data, dry_run=dry_run)
+
+    # Adding image ids
+    if img_import_result.totals["new"]:
+        img_lookup = {
+            row["img_filename"]: row["id"]
+            for row in img_data.dict
+            if row.get("img_filename")
+        }
+        dictionary_entry_data.append_col(
+            [""] * len(dictionary_entry_data), header="related_images"
+        )
```

Comment:
Immutable data is always easier to deal with than having a function modify passed-in data in place. (Easier maintenance, fewer mystery bugs.) Instead of passing in and modifying the dictionary_entry_data, could we have these media import methods return dictionaries for looking up the ID by filename? Then we could pass those to the dictionary entry import method? I haven't looked at this as much as you, so I am probably missing some detail. Happy to chat about it on a call.

Reply:
That makes sense and I agree on the improvement in maintenance. I'll refactor to add in these changes.